dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/compiler/utils/JspFileManager.java

```diff
@@ -76,16 +76,23 @@ public JspFileManager(JavaFileManager fileManager, ClassLoader classLoader) {
         }
     }
 
+    // May be needed for Java Modules? Requires JDK 9+
```

Comment:
Was an issue ever created? I thought this work was done as part of this issue already?

Reply:
https://github.com/OpenLiberty/open-liberty/issues/26090

Reply:
I'd like to discuss this with you. What happens currently with JDK 9+ without this?

Reply:
I think you're fixing this for now with the else if below correct?

Reply:
I'm not sure if `listLocationsForModules` is necessary later on, but I'll leave the comment in just in case.  The main change required here is the one below: 

```
 else if (JAVA_MAJOR_VERSION >= 9 && location.getName().equals("SYSTEM_MODULES[java.base]")) // base module for 9+
            results = super.list(location, packageName, kinds, recurse);
  ```

Reply:
Yes: the fix is in the code under: `else if (JAVA_MAJOR_VERSION >= 9 && location.getName().equals("SYSTEM_MODULES[java.base]")) // base module for 9+`

Reply:
Or I could just remove listLocationsForModules